File: AoC2016/JavaScript/day3-2.js
//puzzle setup
const fs = require('fs');
var input = fs.readFileSync('../day3in.txt','utf8');
input = input.split(/\r?\n/);

//do stuff
for (i in input) input[i] = input[i].trim().replace(/[ \t]{2,}/g,' ').split(' ').map(Number);
var redo = [];
for (let i = 0; i < input.length; i+=3)
{
    let x = [];
    let y = [];
    let z = [];
    for (let j = 0; j < 3; j++)
    {
        x.push(input[i+j][0]);
        y.push(input[i+j][1]);
        z.push(input[i+j][2]);
    }
    redo.push(x);
    redo.push(y);
    redo.push(z);
}
ct = 0;
input = redo;
for (i in input)
{
    let a = input[i][0];
    let b = input[i][1];
    let c = input[i][2];
    if (a + b > c && a + c > b && b + c > a) ct++;
}
console.log(ct);

File: AoC2019/Python3/day3-2.py
#puzzle setup
with open('./day3in.txt','r') as pInput:
    pInput = pInput.read().splitlines()

#do stuff
wireA = pInput[0].split(',')
wireB = pInput[1].split(',')
wacoords = [(0,0)]
wbcoords = [(0,0)]
dirs = 'UDLR'
dircoords = [(0,1),(0,-1),(-1,0),(1,0)]

def wireStep(go,wire):
    for y in range(0,go):
        zips = zip(wire[-1],dircoords[dirs.index(x[:1])])
        wire.append(tuple([sum(zipped) for zipped in zips]))

for x in wireA:
    wireStep(int(x[1:]),wacoords)

for x in wireB:
    wireStep(int(x[1:]),wbcoords)

intersect = list(set(wacoords).intersection(set(wbcoords)))
intersect.remove((0,0))

iSteps = []
for x in intersect:
    stepA = wacoords.index(x)
    stepB = wbcoords.index(x)
    iSteps.append(stepA+stepB)

print(min(iSteps))
